src/services/jobService.ts

```diff
@@ -0,0 +1,140 @@
+import axios, { type AxiosInstance, type AxiosResponse } from "axios";
+import type { JobRole } from "../models/job-role.js";
+
+/**
+ * API Service for handling job-related HTTP requests
+ */
+export class JobService {
```

Comment:
that's a lot of fancy stuff - timeouts, interceptors, logging, error handling... which is great, but let's keep it simple for now so that it's easier to grasp what's happening